test/test.js

```diff
@@ -3680,4 +3681,359 @@ describe('Julia grammar', function () {
+            },
+            {
+                value: '\n',
+                scopes: ['keyword.other.testitem.julia']
+            },
+            {
+                value: 'end',
+                scopes: ['keyword.other.testitem.julia', 'keyword.other.testitem.end.julia']
+            },
+            // Goes back after:
+            {
+                value: '\ny',
+                scopes: []
+            },
+            {
+                value: '=',
+                scopes: ['keyword.operator.update.julia']
+            },
+            {
+                value: '2',
+                scopes: ['constant.numeric.julia']
+            },
+        ]
+        compareTokens(tokens, true_tokens)
+    })
+    it("tokenizes @testitem with nested content", function () {
+        const tokens = tokenize(grammar,
+            dedent(`
+                f(x) = x
+                @testitem "bar" begin
+                    using MyPkg: f
+                    for i in 1:10
+                        @test f(i) == i
+                    end
+                end
+            `)
+        )
+        const true_tokens = [
+            {
+                'value': 'f',
+                'scopes': ['entity.name.function.julia']
+            },
+            {
+                'value': '(',
+                'scopes': ['meta.bracket.julia']
+            },
+            {
+                'value': 'x',
+                'scopes': []
+            },
+            {
+                'value': ')',
+                'scopes': ['meta.bracket.julia']
+            },
+            {
+                'value': ' ',
+                'scopes': []
+            },
+            {
+                'value': '=',
+                'scopes': ['keyword.operator.update.julia']
+            },
+            {
+                'value': ' x\n',
+                'scopes': []
+            },
+            {
+                'value': '@testitem',
+                'scopes': ['keyword.other.testitem.julia', 'keyword.other.testitem.begin.julia']
+            },
+            {
+                'value': ' ',
+                'scopes': ['keyword.other.testitem.julia']
+            },
+            {
+                'value': '"',
+                'scopes': ['keyword.other.testitem.julia', 'string.quoted.double.julia', 'punctuation.definition.string.begin.julia']
+            },
+            {
+                'value': 'bar',
+                'scopes': ['keyword.other.testitem.julia', 'string.quoted.double.julia']
+            },
+            {
+                'value': '"',
+                'scopes': ['keyword.other.testitem.julia', 'string.quoted.double.julia', 'punctuation.definition.string.end.julia']
+            },
+            {
+                'value': ' ',
+                'scopes': ['keyword.other.testitem.julia']
+            },
+            {
+                'value': 'begin',
+                'scopes': ['keyword.other.testitem.julia', 'keyword.control.julia']
```

Comment:
Note how everything is now within this `keyword.other.testitem.julia` **block** of code, so you could potentially change the highlighting of it.